src/core/backbone_router/local.cpp

```diff
@@ -0,0 +1,258 @@
+                 Get<BackboneRouter::Leader>().GetServer16() == Get<Mle::MleRouter>().GetRloc16()
+                 /* If the device itself should be BBR. */
+    );
+
+    serverData.SetSequenceNumber(mSequenceNumber);
+    serverData.SetReregistrationDelay(mReregistrationDelay);
+    serverData.SetMlrTimeout(mMlrTimeout);
+
+    SuccessOrExit(error = Get<NetworkData::Local>().AddService(
+                      THREAD_ENTERPRISE_NUMBER, &serviceData, sizeof(serviceData), true,
+                      reinterpret_cast<const uint8_t *>(&serverData), sizeof(serverData)));
+
+    mIsServiceAdded = true;
+    Get<NetworkData::Local>().SendServerDataNotification();
+
+    otLogInfoNetData("BBR Service added: seqno (%d), delay (%ds), timeout (%ds)", mSequenceNumber, mReregistrationDelay,
+                     mMlrTimeout);
+
+exit:
+    otLogInfoNetData("Add BBR Service: %s", otThreadErrorToString(error));
+    return error;
+}
+
+otError Local::RemoveService(void)
+{
+    otError error       = OT_ERROR_NONE;
+    uint8_t serviceData = NetworkData::ServiceTlv::kServiceDataBackboneRouter;
+
+    SuccessOrExit(
+        error = Get<NetworkData::Local>().RemoveService(THREAD_ENTERPRISE_NUMBER, &serviceData, sizeof(serviceData)));
+
+    mIsServiceAdded = false;
+
+    if (Get<Mle::MleRouter>().IsAttached())
+    {
+        Get<NetworkData::Local>().SendServerDataNotification();
+    }
+
+exit:
+    otLogInfoNetData("Remove BBR Service %s", otThreadErrorToString(error));
+    return error;
+}
+
+void Local::SetState(BackboneRouterState aState)
+{
+    VerifyOrExit(mState != aState);
+
+    mState = aState;
+    Get<Notifier>().Signal(OT_CHANGED_THREAD_BACKBONE_ROUTER_STATE);
+
+exit:
+    return;
+}
+
+void Local::UpdateBackboneRouterPrimary(Leader::State aState, const BackboneRouterConfig &aConfig)
+{
+    OT_UNUSED_VARIABLE(aState);
+
+    VerifyOrExit(mState != OT_BACKBONE_ROUTER_STATE_DISABLED && Get<Mle::MleRouter>().IsAttached());
+
+    // Wait some jitter before trying to Register.
+    if (aConfig.mServer16 == Mac::kShortAddrInvalid)
+    {
+        uint8_t delay = 1;
+
+        if (Get<Mle::MleRouter>().GetRole() != OT_DEVICE_ROLE_LEADER)
+        {
+            delay += Random::NonCrypto::GetUint8InRange(0, mRegistrationJitter < 255 ? mRegistrationJitter + 1
+                                                                                     : mRegistrationJitter);
+        }
+
+        // Here uses the timer resource in Mle.
+        Get<Mle::MleRouter>().SetBackboneRouterRegistrationDelay(delay);
+    }
+    else if (aConfig.mServer16 != Get<Mle::MleRouter>().GetRloc16())
+    {
+        RemoveService();
+
+        if (mState == OT_BACKBONE_ROUTER_STATE_PRIMARY)
+        {
+            // Here increases sequence number if becomes Secondary from Primary.
+            mSequenceNumber++;
+            Get<Notifier>().Signal(OT_CHANGED_THREAD_BACKBONE_ROUTER_LOCAL);
+        }
+
+        SetState(OT_BACKBONE_ROUTER_STATE_SECONDARY);
```

Comment:
```suggestion
        Reset(); 
```
This is effectively same as `Reset();` ?